subsys/bluetooth/host/att.c

```diff
@@ -570,38 +652,45 @@ static int bt_att_chan_send(struct bt_att_chan *chan, struct net_buf *buf)
 	BT_DBG("chan %p flags %lu code 0x%02x", chan, atomic_get(chan->flags),
 	       ((struct bt_att_hdr *)buf->data)->code);
 
+	if (IS_ENABLED(CONFIG_BT_EATT) &&
+	    !att_chan_matches_chan_opt(chan, bt_att_tx_meta_data(buf)->chan_opt)) {
+		return -EINVAL;
+	}
+
 	return chan_send(chan, buf);
 }
 
 static void att_send_process(struct bt_att *att)
 {
-	struct bt_att_chan *chan, *tmp;
-	struct net_buf *buf;
-	int err = -ENOENT;
-
-	buf = net_buf_get(&att->tx_queue, K_NO_WAIT);
-	if (!buf) {
-		return;
-	}
+	struct bt_att_chan *chan, *tmp, *prev = NULL;
+	int err = 0;
 
 	SYS_SLIST_FOR_EACH_CONTAINER_SAFE(&att->chans, chan, tmp, node) {
-		err = bt_att_chan_send(chan, buf);
-		if (err >= 0) {
-			break;
+		if (err == -ENOENT && prev &&
+		    (atomic_test_bit(chan->flags, ATT_ENHANCED) ==
```

Comment:
I don't understand how this condition makes sense. Do you mean to test another channel in the second `test_bit` ?
edit: same comment for `att_req_send_process`

Reply:
Yes, should be `prev->flags` instead of `chan->flags` twice.